src/core/thread/topology.hpp

```diff
@@ -73,6 +73,9 @@ class Neighbor
      *
      */
     enum State
+#if _WIN32
```

Comment:
We might be able to address this with an `OT_PLAT` preprocessor macro... Perhaps something like `OT_PLAT_ENUM_CONTAINER(unsigned int)`. It would be defined to be `: type` on WIN32 and nothing on other platforms.